code/gui.py

```diff
@@ -0,0 +1,240 @@
+#########################################################################
+################################## Clusterer ############################
+clusterer_frame = tk.Frame(master=window)
+clusterer_frame.grid(row=1, column=0, padx=10, pady=10, sticky="nswe")
+
+clusterer = tk.Label(master=clusterer_frame, text="Clusterer")
+clusterer.grid(row=0, column=0, sticky="w")
+# Looping to populate the labels
+for i, text in enumerate(CLUSTERER_ARGS):
+    label = tk.Label(master=clusterer_frame, text=CLUSTERER_ARGS[i][0])
+    label.grid(row=i+1, column=0, sticky="e")
+
+# Argument 1
+def open_file_name():
+    """Open a file."""
+    filepath = askopenfilename(
+        filetypes=FILE_TYPE
+    )
+    if not filepath:
+        return
+    filepath = filepath.rsplit('.', 1)
+    file_name_ent.delete("0", tk.END)
+    file_name_ent.insert(tk.END, filepath[0])
+
+file_name_ent = tk.Entry(master=clusterer_frame, width=60)
+file_name_ent.grid(row=1, column=1, sticky="w")
+file_name_ent.insert(0, CLUSTERER_ARGS[0][1])
+file_name_btn = tk.Button(
+    master=clusterer_frame, text=LABELS[1], command=open_file_name
+)
+file_name_btn.grid(row=1, column=2, sticky="w")
+
+# Argument 2 Output File Directory
+def output_file_browse():
+    """Open a file."""
+    dirpath = askdirectory()
+    if not dirpath:
+        return
+    output_dir_ent.delete("0", tk.END)
+    output_dir_ent.insert(tk.END, dirpath)
+
+output_dir_ent = tk.Entry(master=clusterer_frame, width=60)
+output_dir_ent.grid(row=2, column=1, sticky="w")
+output_dir_ent.insert(0, CLUSTERER_ARGS[1][1])
+output_dir_browse = tk.Button(
+    master=clusterer_frame, text="Browse...", command=output_file_browse
+)
+output_dir_browse.grid(row=2, column=2, sticky="w")
+
+# Argument 3 Output File Name
+output_name_ent = tk.Entry(
+    master=clusterer_frame, text=CLUSTERER_ARGS[2], width=60
+)
+output_name_ent.grid(row=3, column=1, sticky="w")
+output_name_ent.insert(0, CLUSTERER_ARGS[2][1])
+
+# Argument 4 v_scalar
+v_scalar_ent = tk.Entry(master=clusterer_frame)
+v_scalar_ent.grid(row=4, column=1, sticky="w")
+v_scalar_ent.insert(0, CLUSTERER_ARGS[3][1])
+
+# Argument 5
+constrain_partition_var = tk.StringVar(clusterer_frame)
+constrain_partition_var.set(CLUSTERER_ARGS[4][1])
+constrain_partition_menu  = tk.OptionMenu(
+    clusterer_frame, constrain_partition_var, *BOOL_LABELS
+)
+constrain_partition_menu.grid(row=5, column=1, sticky="w")
+
+# Argument 6
+constrain_pcs_var = tk.StringVar(clusterer_frame)
+constrain_pcs_var.set(CLUSTERER_ARGS[5][1])
+constrain_pcs_menu  = tk.OptionMenu(
+    clusterer_frame, constrain_pcs_var, *BOOL_LABELS
+)
+constrain_pcs_menu.grid(row=6, column=1, sticky="w")
+
+def run_clusterer():
+    # returns an error message if first argument is blank
+    if not file_name_ent.get():
+        return messagebox.showwarning(
+            title="Error...", message="Missing Required Argument"
+        )
+
+    output = output_dir_ent.get() + output_name_ent.get()
```

Comment:
This will fail if the string in `output_dir_ent` doesn't end in `/`. When the user chooses their own folder via browse, it doesn't end in `/`, so it will end up saving with a weird file name in the parent directory of whatever directory they chose. You can fix this by using `os.path.join` to combine the two strings, rather than `+`.